crypto/asn1/a_object.c

```diff
@@ -194,6 +194,10 @@ int i2a_ASN1_OBJECT(BIO *bp, const ASN1_OBJECT *a)
             ERR_raise(ERR_LIB_ASN1, ERR_R_MALLOC_FAILURE);
             return -1;
         }
+        if (i > INT_MAX - 1) {  /* catch an integer overflow */
```

Comment:
This check is too late since we already calculated `i+1` when calling `OPENSSL_malloc`.  It also leaks `p` (which could be very large).  Both would be resolved by moving the check to just before the malloc call.

Reply:
Yep, done.